znvis/particle/vector_field.py

```
@@ -117,8 +117,10 @@ def construct_mesh_list(self):
                 self.position = self.position[np.newaxis, :, :]
                 self.direction = self.direction[np.newaxis, :, :]
 
-        except ValueError:
-            raise ValueError("There is no data for this vector field.")
+        except IndexError:
+            raise IndexError("The provided data has an incompatible shape.") from None
```

Ruff (0.13.3)

Avoid specifying long messages outside the exception class

(TRY003)